pkg/build/nodeimage/buildcontext.go

```diff
@@ -314,16 +326,29 @@ func (c *buildContext) createBuildContainer() (id string, err error) {
 	// and a little random bits in case we have multiple builds simultaneously
 	random := rand.New(rand.NewSource(time.Now().UnixNano())).Int31()
 	id = fmt.Sprintf("kind-build-%d-%d", time.Now().UTC().Unix(), random)
+	runArgs := []string{
+		"-d", // make the client exit while the container continues to run
+		// the container should hang forever, so we can exec in it
+		"--entrypoint=sleep",
+		"--name=" + id,
+		"--platform=" + dockerBuildOsAndArch(c.arch),
+		"--security-opt", "seccomp=unconfined", // ignore seccomp
+		"--network=host",
```

Comment:
this was not here before, was it?

Reply:
The line `--network=host` is added by me, the others were there before. The purpose of `--network=host` is to be able to use local host proxy like "http://127.0.0.1:1080".

Reply:
you can not submit a PR with multiple changes without informing the reviewers, this is a change in behavior, it requires additional discussion, and in this case it seems that is only to fit your personal workflow

Reply:
please do only one change at a time

Reply:
> The purpose of --network=host is to be able to use local host proxy like "http://127.0.0.1:1080/".

@Zumium 
This is not necessary for the current PR, you can still use the proxy even without the host network, doesn't it?

Reply:
> The line --network=host is added by me, the others were there before. The purpose of --network=host is to be able to use local host proxy like "http://127.0.0.1:1080/".

We should drop this, because you can't run node containers this way anyhow and will need a different solution for proxy?

Reply:
@liangyuanpeng Well, I'm afraid it's necessary especially for a proxy running on localhost. Imagine your proxy server is running locally on host, for example http://127.0.0.1:1080, which I believe is a common case. If the address is directly set to the container and not using host network, "127.0.0.1" will refer to the container's localhost, and the programs inside the container will fail to connect to proxy because they're not connecting to the right place.

Reply:
@BenTheElder I believe that the core of the problem is the meaning of "localhost" changes inside & outside of the container, and proxies usually bind on localhost for safety reason, preventing a process inside container from connecting to it directly. If host network is not accepted, we may have to add some kind of relay-er, and it will be a more complex solution.

Reply:
you can use the IP of your host, proxies use to listen in all host ip addresses, is easier to modify your proxy than force this pod to run with hostnetwork

Reply:
It's also ok to just keep KIND simple and leave the worry to users. I'll modify my code later.

Reply:
I don't disagree, but this was this way for years and everytime we change something that sounds simple, it always have consequences